### PR TITLE
feature: moved billing_calculate to a laravel command.

### PR DESCRIPTION
**Pull request by kennysinkeler**
Please give a short description what your pull request is for
Moved the billing_calculate.php to a laravel command and new code to billing_calculate.php for legacy purpose.
DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: app/Console/Commands/BillingCalculate.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\Bill;
+use App\Models\BillHistory;
+use Illuminate\Console\Command;
+use LibreNMS\Util\Number;
+
+require_once base_path() . '/includes/init.php';
```
Comment (Jellyfrog) [inline app/Console/Commands/BillingCalculate.php:10]: Sorry this needs to be removed

File: app/Console/Commands/BillingCalculate.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            $i = 0;
+            while ($i <= 24) {
+                unset($class);
+                unset($rate_data);
+                $day_data = getDates($bill->bill_day, $i);
+
+                $datefrom = $day_data['0'];
+                $dateto = $day_data['1'];
+                $check = BillHistory::where(['bill_id' => $bill->bill_id, 'bill_datefrom' => $datefrom, 'bill_dateto' => $dateto])->first();
+                $period = getPeriod($bill['bill_id'], $datefrom, $dateto);
```
Comment (Jellyfrog) [inline app/Console/Commands/BillingCalculate.php:61]: You can port these functions to class based ones, and if they are used in other places just replace the original function with a `return myNewFunction()`

we done that in many places.

Comment (Jellyfrog) [reply app/Console/Commands/BillingCalculate.php:61]: This exact function probably should be in a model tho

File: app/Console/Commands/BillingCalculate.php (hunk 1 above)
Comment (kennysinkeler) [reply app/Console/Commands/BillingCalculate.php:10]: Do you mean this line?(so this functions need also converted to laravel?

```
use LibreNMS\Util\Number;
require_once base_path() . '/includes/init.php';
```

This file is required for commands
```
use Illuminate\Console\Command;

```

File: app/Console/Commands/BillingCalculate.php [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                $date_updated = str_replace('-', '', str_replace(':', '', str_replace(' ', '', $check['updated'])));
```
Comment (mpikzink) [inline app/Console/Commands/BillingCalculate.php:62]: only a hint: ```str_replace([' ', ':', '-'], '', $check['updated'])``` is easier to read

Comment (kennysinkeler) [reply app/Console/Commands/BillingCalculate.php:62]: Thanks your input! I changed this line :)

File: app/Console/Commands/BillingCalculate.php (hunk 1 above)
Comment (Jellyfrog) [reply app/Console/Commands/BillingCalculate.php:10]: Yes and within there lies the pain ;)

Comment (murrant) [reply app/Console/Commands/BillingCalculate.php:10]: Can this line be removed now?

**Comment by murrant** [comment]
Closing due to inactivity.

**Pull request closed without merging**